### ISSUE TITLE
[Bug]: Can't save photos in ZZZ

### ISSUE THREAD

**Issue by Brawl345** [PlayCover/PlayCover#1635]
### Describe the bug

I can't save photos in ZZZ even though I granted access to the album. Read on Discord that others have this problem too but no one reported it here?

Using Playcover 3.0.0 (836)

### Steps to reproduce

1. Try to take a photo in ZZZ
2. Accept the permission request and grant full access to photos
3. The message "Please grant access to your album in the settings" is shown and the photo isn't saved

![Screenshot](https://github.com/user-attachments/assets/935de677-0791-479e-aba0-7508e094e9f4)

### Expected behaviour

The photo should be saved (it works in Genshin and Honkai Star Rail)

### What version of PlayCover are you using?

Nightly/beta (please specify build number)

### What version of macOS are you using?

Sonoma (macOS 14)

### Issue Language

- [X] Yes my issue is written in English

**Comment by MasterJ93** [comment]
I can confirm that this also happens for me. It's been like this since version 1.0.

**Comment by CheneyNine** [comment]
Yes, this also happens for me when I was playing the Star Rail. I thought the photos was saved in the Mac Photo APP, but there was nothing and the Playcover quit by itself.

**Comment by Brawl345** [comment]
No that's a different problem. Photos can be saved in Star Rail but you have to right click on the app -> "Open app data" and navigate to "Documents" and your photos are saved there (/Users/USERNAME/Library/Containers/com.HoYoverse.hkrpgoversea/Data/Documents).

In ZZZ, photos aren't even saved because the app thinks the permissions weren't granted.

**Comment by CheneyNine** [comment]
这是邮箱的自动回复邮件。文件发送成功，我会及时接收，谢谢！

**Comment by CheneyNine** [comment]
> No that's a different problem. Photos can be saved in Star Rail but you have to right click on the app -> "Open app data" and navigate to "Documents" and your photos are saved there (/Users/USERNAME/Library/Containers/com.HoYoverse.hkrpgoversea/Data/Documents).
> 
> In ZZZ, photos aren't even saved because the app thinks the permissions weren't granted.


So sorry for misunderstanding your problem, and very thanks for your help :)

**Comment by RellikJaeger** [comment]
I’m experiencing the same permission issue in Genshin Impact after deleting an already allowed and working "Photos Library.photoslibrary". The Genshin Impact wrapper app isn’t recognizing the newly created Photos Library or its permissions.

I’ve tried allowing and denying all permissions, clearing the app’s preferences, and resetting the system permission scope using `tccutil reset Photos com.miHoYo.GenshinImpact`. While the system permissions appear to be working correctly, the in-game camera no longer triggers a new permission request dialog and doesn’t recognize the already granted permission in system settings.

In the past, I completely uninstalled and reinstalled Genshin Impact, and it resolved the issue. However, this time I would prefer not to redownload everything. I’m unsure how to reset the permission for the in-game camera. Is this handled within the app wrapper or in PlayCover’s configuration?

PlayCover should handle the permissions correctly with the current system permissions.

I'm using PlayCover Version 3.1.0 (856) on M2 MacBook Pro macOS Sequoia Version 15.3.1.

**Comment by CheneyNine** [comment]
这是邮箱的自动回复邮件。文件发送成功，我会及时接收，谢谢！

**Comment by RellikJaeger** [comment]
> I’m experiencing the same permission issue in Genshin Impact after deleting an already allowed and working "Photos Library.photoslibrary". The Genshin Impact wrapper app isn’t recognizing the newly created Photos Library or its permissions.
> 
> I’ve tried allowing and denying all permissions, clearing the app’s preferences, and resetting the system permission scope using `tccutil reset Photos com.miHoYo.GenshinImpact`. While the system permissions appear to be working correctly, the in-game camera no longer triggers a new permission request dialog and doesn’t recognize the already granted permission in system settings.
> 
> In the past, I completely uninstalled and reinstalled Genshin Impact, and it resolved the issue. However, this time I would prefer not to redownload everything. I’m unsure how to reset the permission for the in-game camera. Is this handled within the app wrapper or in PlayCover’s configuration?
> 
> PlayCover should handle the permissions correctly with the current system permissions.
> 
> I'm using PlayCover Version 3.1.0 (856) on M2 MacBook Pro macOS Sequoia Version 15.3.1.

I just found the cause and solution. The problem was that the newly created "Photos Library.photoslibrary" wasn’t set as the System Photo Library, even though the Photos app was working just fine. The solution: Open the Photos app, go to Settings > General, and click on "Use as System Photo Library." This resolved the issue for me.

**Comment by rolling-one** [comment]
Any solutions yet?

**Comment by CheneyNine** [comment]
这是邮箱的自动回复邮件。文件发送成功，我会及时接收，谢谢！
